feat: Update Dart, Flutter API links and improve document structure

mbfakourii opened the pull request:
There was a problem with the flutter `API Reference`

https://docs.parseplatform.org/flutter/guide/

mbfakourii committed:
fix flutter api

mbfakourii committed:
fix flutter api

mtrezza commented on flutter.md:
```diff
@@ -5,7 +5,7 @@ layout: guide
 language: dart
 display_platform: Flutter
-api_reference: https://parse-community.github.io/Parse-SDK-Flutter/api
+api_reference: https://pub.dev/documentation/parse_server_sdk_flutter/latest/
```
I think the API reference would be: https://pub.dev/documentation/parse_server_sdk_flutter/latest/flutter_parse_sdk_flutter/flutter_parse_sdk_flutter-library.html

But actually I think we should generate our own docs and publish them like we do with our other products, instead of redirecting to a 3rd party website.

mbfakourii replied:
Yes, I think the same

But I think it is better to publish in this way until the api is created

mbfakourii replied:
I made the changes

mtrezza commented:
I've added docs publishing to the Flutter SDK; the paths are:
- http://parseplatform.org/Parse-SDK-Flutter/dart/
- http://parseplatform.org/Parse-SDK-Flutter/flutter/

These are the roots of the generated docs, but since we don't need to land on the README, I'd suggest to link directly to the API docs:

- http://parseplatform.org/Parse-SDK-Flutter/dart/flutter_parse_sdk/flutter_parse_sdk-library.html
- http://parseplatform.org/Parse-SDK-Flutter/flutter/flutter_parse_sdk_flutter/flutter_parse_sdk_flutter-library.html

Since we have 2 API docs now, use the folding-widget like in the ObjC guides section:
<img width="283" alt="image" src="https://github.com/parse-community/docs/assets/5673677/6a563d9b-0581-45ae-a2ac-05cb3497545a">

mbfakourii committed:
add api page

mbfakourii committed:
add api page

mbfakourii commented:
Added api.html page to solve the problem

![image](https://github.com/parse-community/docs/assets/20955005/0b948150-8ccf-4702-95b2-33d97b3c85f2)

mbfakourii commented:
> > Added api.html page to solve the problem
> 
> What is the purpose of this extra page? I don't think a page with just 2 links is needed as our current structure throughout the website is to look at them as 2 separate SDKs. In addition, if the Flutter SDK docs include the Dart SDK classes as you mentioned previously, this page makes even less sense I think?

What do you suggest to link only the Flutter version? Or should we add two versions of Dart to the document?

mbfakourii commented:
I don't think two documents are good because they are 90% alike!

mbfakourii commented:
See [this page](https://github.com/parse-community/docs/tree/f2b7788d279de705b60c4263dab74486c1aa5d65/_includes/common)

The idea of the `common` folder can be reasonable, but I think it should have another name

mtrezza commented:
What are these 10% difference? Is that within 1 section or scattered across the docs?

mbfakourii committed:
fix flutter svg icon

mbfakourii committed:
fix flutter svg icon

mbfakourii commented:
> What are these 10% difference? Is that within 1 section or scattered across the docs?
> 
> I should note that we are thinking about combining the docs into a single document. Then in every section there will be tabs to switch between languages / SDKs for the code examples. There is too much duplicate content across the SDKs. But that is a separate, more complex effort. I just say this because it may make the current solution obsolete, so that we don't over-complicate things.

I said about 90 percent. The difference that may be in the settings is the path of the library that is in the [getting-started.md](https://github.com/parse-community/docs/blob/f2b7788d279de705b60c4263dab74486c1aa5d65/_includes/flutter/getting-started.md)

To be precise, maybe 99% are similar!

mtrezza commented:
So there's 1% difference? Could you point out what this 1% is specifically? It may be reasonable to combine the documentation for both into one.

mbfakourii commented:
> You say 99% are similar, so there's only 1% difference, that is content that is in the Flutter docs but not in the Dart docs? Could you point out what this 1% is specifically?
> 
> It may be reasonable to combine the documentation for both into one and not distinguish between the two SDKs on a high level.

This 1% is related to Dart and Flutter library path which is at the beginning of the getting-started.md file

mbfakourii commented:
> You say 99% are similar, so there's only 1% difference, that is content that is in the Flutter docs but not in the Dart docs? Could you point out what this 1% is specifically?
> 
> It may be reasonable to combine the documentation for both into one and not distinguish between the two SDKs on a high level.

This 1% is related to Dart and Flutter library path which is at the beginning of the getting-started.md file

mbfakourii commented:
> The package URL is the only docs difference between the two SDKs? I understand that the Flutter SDK is built on the Dart SDK, but I assume there are other things that the Flutter SDK adds on top of the Dart SDK that would be documented?

Yes, I noticed this when I was merging the documents . Maybe there are things in Flutter that are not documented !

mtrezza commented:
Are you referring to the generated docs (dart doc), or the guide?

mbfakourii commented:
> Are you referring to the generated docs (dart doc), or the guide?

There are two guides available in repository Flutter for [dart](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/packages/dart/README.md) and [flutter](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/packages/flutter/README.md)

mtrezza commented:
Is there a "dart guide"? I can only find a Flutter guide https://docs.parseplatform.org/flutter/guide/

mbfakourii committed:
add dart

mbfakourii commented:
I added the dart

![image](https://github.com/parse-community/docs/assets/20955005/951d2230-2ea2-4028-8da1-96c518b0ae7b)

and deleted the `api.html` page

I noticed that the `push-notifications.md` page is not supported in Dart, so I did not add it to the Dart pages

API Reference paths were also set correctly

I think it is in the best condition now

mbfakourii committed:
Dart base placement

mtrezza commented:
The changes need to be on the front page as well. The _docs.parseplatform.org_ page seems outdated and I mentioned I'm unsure whether it's used anywhere at all.

mbfakourii committed:
Merge branch 'gh-pages' into fix_flutter_api

mbfakourii commented:
@mtrezza 
I think the front page is in another repository

mbfakourii committed:
fix JavaScript

mbfakourii commented:
fixed ! please check [#48](https://github.com/parse-community/parse-community.github.io/pull/48)

mtrezza reviewed (commented):
This looks good, can we merge?

mbfakourii commented:
> This looks good, can we merge? 

Yes
